modules/src/ics07_tendermint/client_def.rs

```diff
@@ -25,17 +47,84 @@ impl ClientDef for TendermintClient {
 
     fn check_header_and_update_state(
         &self,
+        ctx: &dyn ClientReader,
+        client_id: ClientId,
         client_state: Self::ClientState,
         header: Self::Header,
-    ) -> Result<(Self::ClientState, Self::ConsensusState), Error> {
-        if client_state.latest_height() >= header.height() {
-            return Err(Error::low_header_height(
-                header.height(),
-                client_state.latest_height(),
-            ));
+    ) -> Result<(Self::ClientState, Self::ConsensusState), Ics02Error> {
+        // check if a consensus state is already installed; if so it should
+        // match the untrusted header.
+        let header_consensus_state = ConsensusState::from(header.clone());
+        let existing_consensus_state =
+            match maybe_read_consensus_state(ctx, &client_id, header.height())? {
+                Some(cs) => {
+                    // If this consensus state matches, skip verification
+                    // (optimization)
+                    if cs == header_consensus_state {
+                        // Header is already installed and matches the incoming
+                        // header (already verified)
+                        return Ok((client_state, cs));
+                    }
+                    Some(cs)
+                }
+                None => None,
+            };
+
+        let trusted_consensus_state = read_consensus_state(ctx, &client_id, header.trusted_height)?;
+
```

Comment:
we should check that the header revision number matches the `trusted_consensus_state` revision number).

Reply:
This should be taken care of now during deserialization, as per [this commit](https://github.com/informalsystems/ibc-rs/pull/1321/commits/2bbf6f6eb7480dcb8cd2479d1dfd5a22998da4fb).